modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaParameterModelParser.java

```diff
@@ -143,15 +150,65 @@ public ExpressionSupport getExpressionSupport() {
   @Override
   public Optional<ParameterDslConfiguration> getDslConfiguration() {
     if (dslConfiguration == null) {
-      dslConfiguration = parameter.getAnnotation(ParameterDsl.class).map(parameterDsl -> ParameterDslConfiguration.builder()
-          .allowsInlineDefinition(parameterDsl.allowInlineDefinition())
-          .allowsReferences(parameterDsl.allowReferences())
-          .build());
+      Optional<ParameterDsl> legacyAnnotation = parameter.getAnnotation(ParameterDsl.class);
+      Optional<org.mule.sdk.api.annotation.dsl.xml.ParameterDsl> sdkAnnotation =
+          parameter.getAnnotation(org.mule.sdk.api.annotation.dsl.xml.ParameterDsl.class);
+      if (legacyAnnotation.isPresent() && sdkAnnotation.isPresent()) {
+        throw new IllegalParameterModelDefinitionException(format("Parameter '%s' is annotated with '@%s' and '@%s' at the same time",
+                                                                  parameter.getName(),
+                                                                  ParameterDsl.class.getName(),
+                                                                  org.mule.sdk.api.annotation.dsl.xml.ParameterDsl.class
+                                                                      .getName()));
+      } else if (legacyAnnotation.isPresent()) {
```

Comment:
This is a personal opinion, but I think it is better to create a Pojo that contains the info from the annotation, and create the builder with the info from the pojo.